ts/src/program/index.ts

```diff
@@ -304,28 +322,42 @@ export class Program {
    * @param callback  The function to invoke whenever the event is emitted from
    *                  program logs.
    */
-  public addEventListener(
+   public addEventListener(
     eventName: string,
     callback: (event: any, slot: number) => void
-  ): number {
-    const eventParser = new EventParser(this._coder, this._programId);
-    return this._provider.connection.onLogs(this._programId, (logs, ctx) => {
-      if (logs.err) {
-        console.error(logs);
-        return;
-      }
-      eventParser.parseLogs(logs.logs, (event) => {
-        if (event.name === eventName) {
-          callback(event.data, ctx.slot);
+  ): void {
+    if (eventName in this._eventCallbacks){
+      throw new Error(`Event listener for ${eventName} already exists!`);
```

Comment:
Also considered returning a boolean from this function rather than throwing, then clients would have to handle that themselves.